File: backend/InvoiceSystem.Domain/Entities/Invoice.cs
namespace InvoiceSystem.Domain.Entities;

public class Invoice
{
    public long Id { get; set; }

    public long MemberId { get; set; }
    public Member Member { get; set; } = null!;

    public string InvoiceNumber { get; set; } = null!;
    public DateTime InvoiceDate { get; set; }   // 請求日
    public DateTime DueDate { get; set; }       // 支払期限

    public decimal TotalAmount { get; set; }

    public long StatusId { get; set; }
    public InvoiceStatus Status { get; set; } = null!;

    public string? PdfPath { get; set; }
    public string? Remarks { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public ICollection<PaymentAllocation> PaymentAllocations { get; set; } = new List<PaymentAllocation>();
    public ICollection<ReminderHistory> ReminderHistories { get; set; } = new List<ReminderHistory>();

    public ICollection<InvoiceLine> Lines { get; set; } = new List<InvoiceLine>();
}

File: frontend/src/types/invoice.ts
// src/types/invoice.ts

// 既存：古い/簡易モデル（使ってるなら残す）
export interface Invoice {
  id: number;
  memberId: number;
  amount: number;
  status: string;
  invoiceDate: string; // ISO文字列として返るため
  dueDate: string;
}

// 既存：一覧用DTO（現状のまま）
export type InvoiceDto = {
  id: number;
  invoiceNumber: string;
  invoiceDate: string; // ISO 文字列
  dueDate: string;     // ISO 文字列
  totalAmount: number;
  memberName: string;
  statusName: string;
};

// ==============================
// ★追加：詳細画面用DTO（バックエンドの InvoiceDetailDto に合わせる）
// ==============================

export type InvoiceAllocationDto = {
  paymentId: number;
  paymentDate: string;        // ISO
  allocatedAmount: number;

  payerName?: string | null;
  method?: string | null;
  importBatchId?: number | null;
};

export type InvoiceReminderDto = {
  id: number;
  remindedAt: string;         // ISO
  method: string;
  note?: string | null;
};

export type InvoiceDetailDto = {
  id: number;

  memberId: number;
  memberName: string;

  invoiceNumber: string;
  invoiceDate: string;        // ISO
  dueDate: string;            // ISO

  totalAmount: number;
  paidAmount: number;
  remainingAmount: number;

  statusId: number;
  statusName: string;

  pdfPath?: string | null;
  remarks?: string | null;

  createdAt: string;          // ISO

  allocations: InvoiceAllocationDto[];
  reminders: InvoiceReminderDto[];
};
